src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.ts

```diff
@@ -0,0 +1,145 @@
+                pointHoverBorderColor: 'rgba(93,138,201,0)',
+                borderColor: 'rgba(93,138,201,1)',
+                hoverBackgroundColor: 'rgba(93,138,201,1)',
+                hoverBorderColor: 'rgba(93,138,201,1)',
+            },
+            {
+                // Average exercise score bars
+                label: this.exerciseAverageLegend,
+                data: this.exerciseAverageScores.slice(this.currentPeriod, 10 + this.currentPeriod).map((exercise) => exercise.averageScore),
+                type: 'bar',
+                backgroundColor: 'rgba(53,61,71,1)',
+                borderColor: 'rgba(53,61,71,1)',
+                hoverBackgroundColor: 'rgba(53,61,71,1)',
+            },
+        ];
+    }
+
+    private createCharts() {
+        this.barChartOptions = {
+            layout: {
+                padding: {
+                    top: 20,
+                },
+            },
+            responsive: true,
+            hover: {
+                animationDuration: 0,
+            },
+            animation: {
+                duration: 1,
+                onComplete() {
+                    const chartInstance = this.chart,
+                        ctx = chartInstance.ctx;
+                    ctx.textAlign = 'center';
+                    ctx.textBaseline = 'bottom';
+
+                    this.data.datasets.forEach(function (dataset: DataSet, j: number) {
+                        const meta = chartInstance.controller.getDatasetMeta(j);
+                        meta.data.forEach(function (bar: any, index: number) {
+                            const data = dataset.data[index];
+                            ctx.fillText(String(data), bar._model.x, bar._model.y - 5);
+                        });
+                    });
+                },
+            },
+            scales: {
+                yAxes: [
+                    {
+                        ticks: {
+                            beginAtZero: true,
+                            min: 0,
+                            max: 100,
+                        },
+                    },
+                ],
+                xAxes: [
+                    {
+                        gridLines: {
+                            offsetGridLines: false,
+                        },
+                        ticks: {
+                            autoSkip: false,
+                            callback(title: string) {
+                                return title.length > 10 ? title.substr(0, 10) + '...' : title;
+                            },
+                        },
+                    },
+                ],
+            },
+        };
+    }
+
+    // handles arrow clicks and updates the exercises which are shwon
+    public switchTimeSpan(index: boolean): void {
+        // eslint-disable-next-line chai-friendly/no-unused-expressions
+        index ? (this.currentPeriod += 1) : (this.currentPeriod -= 1);
+        this.initializeChart();
+    }
```

Comment:
Why does a boolean value called index? And it might be cleaner to write an if statement since you are not using the value of ternary expression or something like this.currentPeriod += index ? 1 : -1; so that you use the value that is returned by the ternary expression. But I think if else is better in this case

Reply:
Changed the statement, i use a boolean here since there are 2 options, either forwards or backwards. I renamed the variable so it should be clearer